### ISSUE TITLE
Design Review 2019-02-13 21:00 UTC (amp-carousel v2)

### ISSUE THREAD

**Issue by mrjoro** [ampproject/amphtml#20535]
Time: [2019-02-13 21:00 UTC](https://www.timeanddate.com/worldclock/meeting.html?year=2019&month=02&day=13&iv=0) ([add to Google Calendar](http://www.google.com/calendar/event?action=TEMPLATE&text=AMP%20Project%20Design%20Review&dates=20190213T210000Z/20190213T220000Z&details=https%3A%2F%2Fhangouts.google.com%2Fhangouts%2F_%2Fgoogle.com%2Famp-eng-desrev))
Location: [Video conference via Google Hangouts](https://hangouts.google.com/hangouts/_/google.com/amp-eng-desrev)

The AMP Project holds weekly engineering [design reviews](https://github.com/ampproject/amphtml/blob/master/contributing/design-reviews.md).  **We encourage everyone in the community to participate in these design reviews.**

If you are interested in bringing your design to design review, read the [design review documentation](https://github.com/ampproject/amphtml/blob/master/contributing/design-reviews.md) and add a link to your design doc by the Monday before your design review.

When attending a design review please read through the designs *before* the design review starts.  This allows us to spend more time on discussion of the design.

We rotate our design review between times that work better for different parts of the world as described in our [design review documentation](https://github.com/ampproject/amphtml/blob/master/contributing/design-reviews.md), but you are welcome to attend any design review.  If you cannot make any of the design reviews but have a design to discuss please let mrjoro@ know on [Slack](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md#discussion-channels) and we will find a time that works for you.

**Comment by sparhami** [comment]
I'd like to discuss #20595, a new version of `<amp-carousel>`

**Comment by prateekbh** [comment]
I'd also like to do a complete design review of AMP-SW along with the release/rollback process.

**Comment by danielrozenberg** [comment]
I'd like to discuss #20744, a method to unblock AMP contributors in the face of flaky tests on Travis

**Comment by nainar** [comment]
Hi @prateekbh and @danielrozenberg just to set expectations the carousel topic proposed by @sparhami might end up taking the full session.

**Comment by aghassemi** [comment]
## `amp-carousel v2` by @sparhami 
- ITI for set of 3 components related to amp-carousel and friends which extend the use of the component and have builtin UI: `base-carousel`, `inline-gallery`, `stream-gallery`
- The idea of `base-carousel` is that it will be flexible, but we like to encourage others to use one of the hight level ones
- Focusing on open questions, 
- one is responsive layouts:
- the main problem is the layout of the element changes between mobile and desktop for inline-gallery. In desktop there is an extra thumbnail area under the carousel but not in mobile case.
- another case where this becomes a problem is captions.
- **Proposal for responsive layout:** expand `layout=responsive` and allow an additional `fixed-height` that gets added to the final height.
- @torch2424 adding an extra seems funny, maybe a new `layout=responsive-fixed-height` also name is confusing since we have a `layout=fixed-height.`
- @lannka We also ran into this with Ads since Ad has a fixed size but we want to add extra UI to it.
- @sparhami to follow up with an new issue.
- @cramforce `additional-height` might be a better name.
- @dvoytenko would the additional height go to top or bottom. Another option is doing this layout in CSS. @cramforce if this is purely possible in CSS, there is no reason to add an attribute. We can just do an ABE sample
-  **carousel API** 
- 1- customization of arrows: proposal is for them to pass button instead of customizing our CSS
- @nainar if we decide to change arrows, is that a breaking change? no, style changes should be fine.
- @dvoytenko `controls` attribute may still be needed so you can bind them. Good idea, we can add that.
- @aghassemi using `slot` is different then current APIs, @sparhami, have it this way for shadow DOM as the new convention even though different from past APIs which were just some random attribute.
- **responsive configuration**
- problem in the past has been needing to create multiple carousel and controlling them with `media`
- in carousel v2, we have responsive attributes which follow same syntax as `sizes`
- @dvoytenko we could also support `figure`, @sparhami we could, involved cloning the caption.
- @aghassemi slot API seems very inflexible @sparhami we can not slot descendants.
- @cathyxz: sometimes swiping the carousel sipes the viewer, @aghassemi it is usually due to padding.
- @aghassemi any cases we can do SSR? @cramforce for now we just do layout, SSRing `amp-img` would be a higher priority.
- **getting the initial index from Url**
- what's a good standard way of doing this?
- VARIABLE substitution is one option.
- @cramforce can we use CSS variables for the responsive attributes instead of current context? @sparhami yeah but CSS variables don't fire any change event. We should talk to Web Platform to add events.
